File: LMS_CMS_Backend/LMS_CMS_DAL/Models/Domains/LMS/AssignmentStudentQuestionAnswerOption.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LMS_CMS_DAL.Models.Domains.LMS
{
    public class AssignmentStudentQuestionAnswerOption : AuditableEntity
    {
        [Key]
        public long ID { get; set; }
        public long Order { get; set; }
        public string Answer { get; set; }

        [ForeignKey("AssignmentStudentQuestion")]
        public long AssignmentStudentQuestionID { get; set; }
        public AssignmentStudentQuestion AssignmentStudentQuestion { get; set; }

        [ForeignKey("QuestionBankOption")]
        public long? SelectedOpionID { get; set; }
        public QuestionBankOption? QuestionBankOption { get; set; }

        [ForeignKey("SubBankQuestion")]
        public long? SubBankQuestionID { get; set; }
        public SubBankQuestion? SubBankQuestion { get; set; }
    }
}

File: LMS_CMS_Frontend/src/app/Models/LMS/assignment-student-question-answer-option.ts
export class AssignmentStudentQuestionAnswerOption {
    constructor(
        public id: number = 0,
        public order: number = 0,
        public assignmentStudentQuestionID: number  = 0,
        public selectedOpionID: number = 0,
        public subBankQuestionID: number = 0,
        public answer: string = "",

    ) { }
}
